File: DotaStats/ClientApp/src/app/HeroCounter.ts

export class HeroCounter{
    name: string;
    SearchedHero: Hero;
    HeroCounters: Array<Hero>;
  }

  export class Hero
    {
        id :Int32Array;
        name:string;
        localized_name: string;
        primary_attr: string;
        attack_type: string;
        roles: Array<string>; 
        legs: Int32Array;
        winRateFavor: string;
    }

File: DotaStats/Controllers/HeroCounter.cs
using System;
using System.Collections.Generic;

namespace DotaStats.Controllers
{
    [Serializable]
    public class HeroCounter
    {
        public string Name { get; set; }

        public Hero SearchedHero { get; set; }

        public IEnumerable<Hero> HeroCounters { get; set; }
    }
}